lib/Listener/GroupMembershipListener.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2020 Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Listener;
+
+use OCA\Talk\Exceptions\ParticipantNotFoundException;
+use OCA\Talk\Manager;
+use OCA\Talk\Model\Attendee;
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCA\Talk\Service\ParticipantService;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\Group\Events\UserAddedEvent;
+use OCP\Group\Events\UserRemovedEvent;
+use OCP\IGroup;
+use OCP\IGroupManager;
+use OCP\IUser;
+
+class GroupMembershipListener implements IEventListener {
+
+	/** @var IGroupManager */
+	private $groupManager;
+	/** @var Manager */
+	private $manager;
+	/** @var ParticipantService */
+	private $participantService;
+
+	public function __construct(IGroupManager $groupManager,
+								Manager $manager,
+								ParticipantService $participantService) {
+		$this->groupManager = $groupManager;
+		$this->manager = $manager;
+		$this->participantService = $participantService;
+	}
+
+	public function handle(Event $event): void {
+		if ($event instanceof UserAddedEvent) {
+			$this->addNewMemberToRooms($event->getGroup(), $event->getUser());
+		}
+		if ($event instanceof UserRemovedEvent) {
+			$this->removeFormerMemberFromRooms($event->getGroup(), $event->getUser());
+		}
+	}
+
+	protected function addNewMemberToRooms(IGroup $group, IUser $user): void {
+		$rooms = $this->manager->getRoomsForActor(Attendee::ACTOR_GROUPS, $group->getGID());
+
+		foreach ($rooms as $room) {
+			try {
+				$participant = $room->getParticipant($user->getUID());
+				if ($participant->getAttendee()->getParticipantType() === Participant::USER_SELF_JOINED) {
+					$this->participantService->updateParticipantType($room, $participant, Participant::USER);
+				}
+			} catch (ParticipantNotFoundException $e) {
+				$this->participantService->addUsers($room, [[
+					'actorType' => Attendee::ACTOR_USERS,
+					'actorId' => $user->getUID(),
+				]]);
```

Comment:
`displayName` is not given here, while it is in [`ParticipantService::addGroup`](https://github.com/nextcloud/spreed/pull/4810/files#diff-8d29629ca6909ff3b00eb3ff13094b704d2b10ce22eff12e6921dccf08a83100R344). I do not really know if it is relevant, but just in case.

Reply:
Fixed, it saves a query later, but was not there when this PR was started and I missed it on the rebase